### PR TITLE
Adding a cancelled field to tell if a cancellable task is cancelled

### PR DESCRIPTION
**Pull request by meghasaik**
Signed-off-by: Megha Sai Kavikondala <kavmegha@amazon.com>

Description
A cancelled field is added which tells if the cancellable task is cancelled. At some points, even if the cancellable task is cancelled it still runs, so to show that the task is cancelled, a cancelled field is shown in the tasks API, cancelled:true .

Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/520

Check List
 New functionality includes testing.
 All tests pass
 New functionality has been documented.
 New functionality has javadoc added
 Commits are signed per the DCO using --signoff
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check here.

### CODE REVIEW

File: server/src/main/java/org/opensearch/tasks/TaskInfo.java [hunk 1]
```diff
@@ -92,9 +95,13 @@ public TaskInfo(
         long startTime,
         long runningTimeNanos,
         boolean cancellable,
+        boolean cancelled,
         TaskId parentTaskId,
         Map<String, String> headers
     ) {
+        if (cancellable == false && cancelled == true) {
+            throw new IllegalArgumentException("uncancellable task cannot be cancelled");
```
Comment (dblock) [inline server/src/main/java/org/opensearch/tasks/TaskInfo.java:103]: I think it should just say "task cannot be cancelled", it implies the former

File: server/src/main/java/org/opensearch/tasks/TaskInfo.java [hunk 2]
```diff
@@ -119,6 +127,14 @@ public TaskInfo(StreamInput in) throws IOException {
         startTime = in.readLong();
         runningTimeNanos = in.readLong();
         cancellable = in.readBoolean();
+        if (in.getVersion().onOrAfter(Version.V_2_0_0)) {
+            cancelled = in.readBoolean();
+        } else {
+            cancelled = false;
+        }
+        if (cancellable == false && cancelled == true) {
+            throw new IllegalArgumentException("uncancellable task cannot be cancelled");
```
Comment (dblock) [inline server/src/main/java/org/opensearch/tasks/TaskInfo.java:136]: task cannot be cancelled

**Review by dblock — APPROVED** [review]
Look through failing gradle checks.

Would change the message, otherwise LGTM. It's still wrong that `TaskInfo` works for multiple kinds of tasks and that `Task` knows about its child. I would refactor that implementation, but that can be a different PR.

**Comment by andrross** [comment]
```
Execution failed for task ':distribution:bwc:bugfix:buildBwcLinuxTar'.
> Building 1.2.1 didn't generate expected file /var/CITOOL/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/search/distribution/bwc/bugfix/build/bwc/checkout-1.2/distribution/ar
chives/linux-tar/build/distributions/opensearch-min-1.2.1-SNAPSHOT-linux-x64.tar.gz
```

I think you need to rebase another version update commit.

**Review by andrross — APPROVED** [review]

**Comment by dblock** [comment]
@meghasaik FYI, there was a regression in this PR, see https://github.com/opensearch-project/OpenSearch/pull/5377

**Pull request merged**